File: src/NotificationHubsSample.WinUsingWebApi (v2)/NotificationHubsSample.WinUsingWebApi (v2).Shared/DeviceInfo.cs
using System.Collections.Generic;

namespace NotificationHubsSample
{
    /// <summary>
    /// Define the DeviceInfo.
    /// </summary>
    public class DeviceInfo
    {
        /// <summary>
        /// Gets or sets the registration identifier.
        /// </summary>
        /// <value>The registration identifier.</value>
        public string RegistrationId { get; set; }

        /// <summary>
        /// Gets or sets the platform.
        /// </summary>
        /// <value>The platform.</value>
        public string Platform { get; set; }

        /// <summary>
        /// Gets or sets the handle.
        /// </summary>
        /// <value>The handle.</value>
        public string Handle { get; set; }

        /// <summary>
        /// Gets or sets the tags.
        /// </summary>
        /// <value>The tags.</value>
        public IEnumerable<string> Tags { get; set; }
    }
}

File: Using Registration Model/Case 2/NotificationHubsSample.Win81-VB/DeviceInfo.vb
''' <summary>
''' Define the DeviceInfo.
''' </summary>
Public Class DeviceInfo
    ''' <summary>
    ''' Gets or sets the registration identifier.
    ''' </summary>
    ''' <value>The registration identifier.</value>
    Public Property RegistrationId() As String
        Get
            Return m_RegistrationId
        End Get
        Set(value As String)
            m_RegistrationId = Value
        End Set
    End Property
    Private m_RegistrationId As String

    ''' <summary>
    ''' Gets or sets the platform.
    ''' </summary>
    ''' <value>The platform.</value>
    Public Property Platform() As String
        Get
            Return m_Platform
        End Get
        Set(value As String)
            m_Platform = Value
        End Set
    End Property
    Private m_Platform As String

    ''' <summary>
    ''' Gets or sets the handle.
    ''' </summary>
    ''' <value>The handle.</value>
    Public Property Handle() As String
        Get
            Return m_Handle
        End Get
        Set(value As String)
            m_Handle = Value
        End Set
    End Property
    Private m_Handle As String

    ''' <summary>
    ''' Gets or sets the tags.
    ''' </summary>
    ''' <value>The tags.</value>
    Public Property Tags() As IEnumerable(Of String)
        Get
            Return m_Tags
        End Get
        Set(value As IEnumerable(Of String))
            m_Tags = Value
        End Set
    End Property
    Private m_Tags As IEnumerable(Of String)
End Class